primitives/io/src/lib.rs

```diff
@@ -410,8 +413,10 @@ pub trait Crypto {
 		self.extension::<KeystoreExt>()
 			.expect("No `keystore` associated for the current context!")
 			.read()
-			.ed25519_key_pair(id, &pub_key)
-			.map(|k| k.sign(msg))
+			.sign_with((*pub_key).into(), id, msg)
+			.map(|sig| sig.try_into().ok())
```

Comment:
I just learned about the nicer `and_then` (instead of `map(..).flatten()` pattern) ...probably something to use here..